docs/components/concepts/workflow-patterns.md

```diff
@@ -0,0 +1,296 @@
+:::
+
+### External messages/events
+
+You might also want to react to certain incoming messages or events in an existing process. A good example is a customer canceling the current order fulfillment process. This might be possible only in a certain process phase and lead to different actions. This is related to [Workflow Pattern 23: Transient Trigger](http://www.workflowpatterns.com/patterns/control/new/wcp23.php) and [Workflow Pattern 24: Persistent Trigger](http://www.workflowpatterns.com/patterns/control/new/wcp24.php).
+
+As with timers, you can leverage [boundary events](/docs/components/modeler/bpmn/events/#boundary-events) or [event subprocesses](/docs/components/modeler/bpmn/event-subprocesses/).
+
+<div bpmn="workflow-patterns/reactive-message-events.bpmn" callouts="subprocess1, event1, event2, event3" />
+
+Assume that an order cancelation message comes in for the current process instance using [message correlation](/docs/components/concepts/messages/).
+
+<span className="callout">1</span>
+
+Subprocesses can be easily used to define phases of a process, as the cancelation is treated differently depending on the current process phase.
+
+<span className="callout">2</span>
+
+For example, a cancelation during the clearing phase has no consequences and can simply be executed.
+
+<span className="callout">3</span>
+
+But when the process is already in the preparation phase it might need to clean up certain things properly.
+
+<span className="callout">4</span>
+
+During delivery, it does not even allow cancelations anymore. This is also why this event is non-interrupting (dashed line), so we keep doing **Delivery**.
+
+### Correlation mechanisms
+
+Mapping external messages to an existing process instance is called [message correlation](/docs/components/concepts/messages/). This is a crucial functionality to ensure you can communicate with process instances from the outside.
+
+There are two main problems to solve:
+
+1. How to find the right process instance? In Camunda, this is solved by a `message name` and a `correlation key` (e.g. `orderCanceled` and `order-42`).
+
+2. How to persist messages if a process instance is not yet ready to receive that message yet? In Camunda, this is solved by having an internal message store and a `time to live` attached to messages. This is related to [Workflow Pattern 24: Persistent Trigger](http://www.workflowpatterns.com/patterns/control/new/wcp24.php)
+
+You can find more information in [our documentation about messages](/docs/components/concepts/messages/).
+
+### Events from subprocesses
+
+Sometimes, you a subprocess need to communicate with its parent without ending the subprocess yet. BPMN allows this by an [escalation event](/docs/components/modeler/bpmn/bpmn-coverage/).
```

Comment:
Do you mean you need to communicate with a parent? Might just be a typo here.

Reply:
```suggestion
Sometimes, a subprocess needs to communicate with its parent process without ending the subprocess yet. BPMN allows this by an [escalation event](/docs/components/modeler/bpmn/bpmn-coverage/).
```

Reply:
Amara is completely right. And yes - typo